xrpl/models/transactions/transaction.py

```diff
@@ -209,7 +209,15 @@ def to_dict(self: Transaction) -> Dict[str, Any]:
         """
         # we need to override this because transaction_type is using ``field``
         # which will not include the value in the objects __dict__
-        return {**super().to_dict(), "transaction_type": self.transaction_type.value}
+        dictionary = {
+            **super().to_dict(),
+            "transaction_type": self.transaction_type.value,
+        }
+        if isinstance(self.flags, list):
+            dictionary["flags"] = 0
+            for flag in self.flags:
+                dictionary["flags"] |= flag
+        return dictionary
```

Comment:
suggestion:

```python
def to_dict(self: Transaction) -> Dict[str, Any]:
  return {**super().to_dict(), "transaction_type": self.transaction_type.value, "flags": self._flags_to_dict()}
  
def _flags_to_dict(self: Transaction) -> int:
  if isinstance(self.flags, int):
    return self.flags
  accumulator = 0
  for flag in self.flags:
    accumulator |= flag
  return accumulator
```

factoring out into helper - just readability imo. take it or leave it